blog/2024-11-linked-signal/README.md

```diff
@@ -0,0 +1,266 @@
+However, the Linked Signal makes it possible to manually set the value in the `overrideFirstBook()` method.
+
+```typescript
+import { Component, linkedSignal, signal } from '@angular/core';
+
+@Component({
+  selector: 'app-book-list',
+  template: `
+    <p>First book in list: {{ firstBook() }}</p>
+    <button (click)="changeBookList()">Change Book List</button>`
+})
+export class BookListComponent {
+  books = signal(['Angular', 'React', 'Vue']);
+
+  firstBook = linkedSignal({
+    source: this.books,
+    computation: books => books[0]
+  });
+
+  overrideFirstBook() {
+    // Manually updating `firstBook`, which now returns 'jQuery'
+    this.firstBook.set('jQuery');
+  }
+
+  changeBookList() {
+    // Changing `books` causes `firstBook` to reset, now returning 'Next.js'
+    this.books.set(['Next.js', 'Svelte', 'Nuxt']);
+  }
+}
+```
+
+In this example:
+- The Linked Signal `firstBook` initially points to the first book in the list of `books`.
+- When `books` changes, `firstBook` recalculates to reflect the first item in the updated list.
+- However, we can also override the value manually, as shown with the 'jQuery' book. 
+
+
+### Use Case with Input Signals
+
+A common use for a Linked Signal could be components that reset based on an input signal. 
+For example, a shopping cart component might want to reset the quantity field when the selected product changes. 
+While we could achieve the same result with `computed`, we also want to be able to set the quantity based on the user's input.
+
+```typescript
+import { Component, input, linkedSignal } from '@angular/core';
+
+@Component({
+  selector: 'app-shopping-cart',
+  template: `
+    <p>Book: {{ selectedBook().title }}</p>
+    <input [(ngModel)]="amount">`,
+  imports: [FormsModule]
+})
+export class ShoppingCartComponent {
+  selectedBook = input.required<Book>();
+  amount = linkedSignal({
+    source: this.selectedBook,
+    computation: () => 1 // Resets to 1 when selectedBook changes
+  });
+}
+```
+
+In this case, whenever `selectedBook` changes, the `amount` input resets to `1`. 
+This pattern is useful in forms where we want fields to reset to a default state when certain inputs change.
+
+## Advanced Scenarios for Linked Signals
+
+### Nested State Management
+
+Suppose you have nested data such as book properties (`title` and `rating`), and we want these fields to reset when a different `book` is selected. 
+Here’s how we could manage this with a Linked Signal:
+
+```typescript
+import { Component, input, linkedSignal } from '@angular/core';
+
+@Component({
+  selector: 'app-book',
+  template: `
+    <p>Title: {{ title() }}</p>
+    <p>Rating: {{ rating() }}</p>
+
+    <button (click)="doRateUp()">Rate up</button>
+  `,
+})
+class BookComponent  {
+  book = input.required<Book>();
+  ratingChange = output<{ isbn: string, newRating: number }>();
+
+  title = computed(() => this.book().title);
+
+  rating = linkedSignal({
```

Comment:
I propose that it would be enough here to write it like the computed above with just a function.